doc/contribute/onboarding.rst

```diff
@@ -0,0 +1,102 @@
+Onboarding
+==========
+
+This page aims to be a list of resources that should be helpful if you want to start contributing to CCF.
+
+Introduction
+------------
+
+Start by reading :doc:`/overview/what_is_ccf`.
+
+If you encounter any terms or acronyms that you don't know, check the :doc:`/overview/glossary`. If the word you are looking for isn't defined, create an `issue <https://github.com/microsoft/CCF/issues/new/choose>`_.
+
+Create an SGX VM if necessary
+-----------------------------
+
+If you intend to make changes that need to work on Intel SGX, you will want an :doc:`SGX VM </contribute/create_vm>` to check you didn't introduce instructions that are illegal in enclave, and to evaluate the performance impact of your change.
+
```

Comment:
I feel like this should be phrased the other way around to lower the barrier to entry: unless the changes are large (e.g. bringing in another library) or require touching SGX-specific features (e.g. attestation), developing on non-SGX machine is OK and the CI will run of SGX-specific hardware anyway. Otherwise, you can create an SGX-VM?

Reply:
For the time being, SGX is the only target with security properties, and it's relatively easy as we've seen to create incompatibilities even when not touching SGX-specific features. Once there are other supported TEEs, we can revisit this.